crypto/x509/v3_purp.c

```diff
@@ -126,16 +164,40 @@ X509_PURPOSE *X509_PURPOSE_get0(int idx)
     return sk_X509_PURPOSE_value(xptable, idx - X509_PURPOSE_COUNT);
 }
 
+X509_PURPOSE *X509_PURPOSE_get0(int idx)
+{
+    X509_PURPOSE *p;
+
+    if (idx < 0)
+        return NULL;
+    if (idx < (int)X509_PURPOSE_COUNT)
+        return xstandard + idx;
+    if (!ensure_init()
+        || !CRYPTO_THREAD_read_lock(xptable_lock))
+        return NULL;
+    p = sk_X509_PURPOSE_value(xptable, idx - X509_PURPOSE_COUNT);
```

Comment:
`get_entry()` ??

Reply:
Part of get_entry is slightly redundant in this case because `idx` has  already been checked, but it doesn't really matter, so OK.